ur_robot_driver/doc/ROS_INTERFACE.md

```diff
@@ -760,6 +760,10 @@ Setup the mounted payload through a ROS service
 
 Calling this service will zero the robot's ftsensor. Note: On e-Series robots this will only work when the robot is in remote-control mode.
 
+##### reset_revolution_counter ([std_srvs/Trigger](http://docs.ros.org/api/std_srvs/html/srv/Trigger.html))
+
+Calling this service will reset the revolution counter for the robot's wrist_3_link. Note: On e-Series robots this will only work when the robot is in remote-control mode.
```

Comment:
Just want to clarify for the future reader that it is only the UR3 that will benefits from this.
```suggestion
The UR3 and UR3e's wrist 3 can do infinite rotations and are therefore not limited to ±2Pi.
Calling this service will reset the revolution counter for the robot's wrist_3_link. Note: On e-Series robots this will only work when the robot is in remote-control mode.
This is handy to avoid rotating back after e.g. a screwing task, and thereby avoid compensating for this in the program.
```

Reply:
Might be nice to make this even stronger and have the service fail if the robot is not a UR3.

Reply:
I suppose something like this:
```c++
bool HardwareInterface::resetRevolutionCounter(std_srvs::TriggerRequest& req, std_srvs::TriggerResponse& res)
{
  if (ur_driver_->getName() == 'UR3e' || ur_driver_->getName() == 'UR3')
  {
    std::mutex mtx;
    handleRobotProgramState(false);
    mtx.lock();
    res.success = this->ur_driver_->sendScript(R"(sec resetRevolutionCounter():
    reset_revolution_counter()
  end
  )");
    mtx.unlock();
    handleRobotProgramState(true);
  }
  else
  {
    std::stringstream ss;
    ss << "Resetting the revolution counter is only available for UR3/e robots. This robot is "
       << ur_driver_->getName();
    ROS_ERROR_STREAM(ss.str());
    res.message = ss.str();
    res.success = false;
  }
  return true;
}
```

Though, I don't know where to get the robot's name...

Reply:
Out of my head I am not sure whether there's something predefined to get the robot model through the API. But that would have been my next point on the list.

@gavanderhoorn wrote:
> Might be nice to make this even stronger and have the service fail if the robot is not a UR3.

I would even have supposed to advertise it only for UR3(e) robots. But that might be more confusing than simply letting the service fail...